warehouse/integrations/snowflake/snowflake.go

```diff
@@ -1108,6 +1127,24 @@ func (sf *Snowflake) connect(ctx context.Context, opts optionalCreds) (*sqlmw.DB
 	return middleware, nil
 }
 
+func ParsePrivateKey(privateKey, passPhrase string) (*rsa.PrivateKey, error) {
```

Comment:
Exporting stuff just because you need it in your test is a small imo. You could simply rename the test package to `snowflake`, I don't think doing the `snowflake_test` thing is needed really.

Reply:
Sure. let me try this. Last time I remember, we used to have a circular dependency.

Reply:
Indeed.

```

Reply:
It's because of the validation package. I had [this](https://linear.app/rudderstack/issue/PIPE-170/validations-cleanup) ticket, a long time back to refactor the validations package to remove Init and Globals but it didn't got prioritized.

Reply:
do we have a utils package where this method can be moved instead of keeping this under snowflake package? there's a possibility of this method to be used in other integrations. wdyt?

Reply:
Since the scope is limited as per Snowflake we are accepting only RSA and not other kinds of private keys like `DSA`, `OPENSSH`, `EC`, etc.. so I would probably keep this under Snowflake. If in future the scope extends we can probably move it to utils.